File: Json/src/main/java/org/elkoserver/json/JsonWrapping.java
package org.elkoserver.json;

public class JsonWrapping {
    public static Object wrapWithElkoJsonImplementationIfNeeded(Object object) {
        Object result;

        if (object instanceof com.grack.nanojson.JsonObject) {
            result = new JsonObject((com.grack.nanojson.JsonObject) object);
        } else if (object instanceof com.grack.nanojson.JsonArray) {
            result = new JsonArray((com.grack.nanojson.JsonArray) object);
        } else {
            result = object;
        }

        return result;
    }
}

File: Json/src/main/kotlin/org/elkoserver/json/JsonWrapping.kt
package org.elkoserver.json

object JsonWrapping {
    fun <TObject : Any?> wrapWithElkoJsonImplementationIfNeeded(`object`: TObject): TObject =
            when (`object`) {
                is com.grack.nanojson.JsonObject -> JsonObject(`object`) as TObject
                is com.grack.nanojson.JsonArray -> JsonArray(`object`) as TObject
                else -> `object`
            }
}
